realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -291,6 +291,25 @@ public boolean isValid() {
      * @param fieldName the field to compare.
      * @param value the value to compare with.
      * @return the query object.
+     * @throws java.lang.IllegalArgumentException if one or more arguments do not match class or field type, or if
+     * {@code fieldName} points to a linked field.
+     */
+    public RealmQuery<E> equalTo(String fieldName, byte[] value) {
```

Comment:
Seems to be missing unit tests for the field name being invalid or not pointing to a byte field?